hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java

```diff
@@ -418,6 +419,30 @@ public static boolean isMetaFile(String p) {
     return p != null && p.endsWith(META_WAL_PROVIDER_ID);
   }
 
+  /**
+   * Comparator used to compare logs together based on their start time.
+   * Just compares start times and nothing else.
+   */
+  public static class WALsStartTimeComparator implements Comparator<Path> {
```

Comment:
WALFileStartTimeComparator is better? We are comparing Paths here.

Reply:
It extracts the timestamp from the filename and compares that so, yes, compares the 'start' logging timestamp.

Reply:
Here is the original comparator named LogComparator:

```
-    @Override
-    public int compare(Path o1, Path o2) {
-      return Long.compare(getTS(o1), getTS(o2));
-    }
-
-    /**
-     * <p>
-     * Split a path to get the start time
-     * </p>
-     * <p>
-     * For example: 10.20.20.171%3A60020.1277499063250
-     * </p>
-     * @param p path to split
-     * @return start time
-     */
-    private static long getTS(Path p) {
-      return AbstractFSWALProvider.getWALStartTimeFromWALName(p.getName());
-    }
```

Reply:
I mean is it better to change 'WALs' to 'WALFile'? Or at least to 'WAL'? 'WALs' looks strange...

Reply:
I am +1 for "WALStartTimeComparator". :-)